torchdata/nodes/samplers/multi_node_weighted_sampler.py

```diff
@@ -63,21 +65,30 @@ def __init__(
         seed: int = 0,
     ) -> None:
         super().__init__()
+
         self.source_nodes = source_nodes
         self.weights = weights
         self.stop_criteria = stop_criteria
         self.dataset_names = list(self.source_nodes.keys())
         self._num_yielded = 0
+        self._started = False
         self.seed = seed
 
+        # Setup rank and world size
         if rank is None or world_size is None:
             self.rank, self.world_size = get_rank_and_world_size()
         else:
             self.rank = rank
             self.world_size = world_size
 
+        self._epoch_random_seed = 0
```

Comment:
Let's just call this `_epoch` for brevity, since it's now roughly what it's trying to capture

Reply:
We do not guarantee if this is exactly the epoch, do we still want to call it _epoch ?